sdk/storage/Azure.Storage.Common/src/Shared/PooledMemoryStream.cs

```diff
@@ -54,13 +54,24 @@ private class BufferPartition
         /// </summary>
         private List<BufferPartition> BufferSet { get; } = new List<BufferPartition>();
 
+        /// <summary>
+        /// Size to rent from MemoryPool during Writes.
+        /// </summary>
+        public int ArrayPoolRentalSize { get; }
```

Comment:
This appears to serve the same purpose as `MaxArraySize`. What is the difference?

Reply:
Nothing I guess, I didn't understand what `MaxArraySize` was from the description.

Reply:
I'll remove in the next iteration.

Reply:
Thanks. Feel free to refactor the name/docstring to be more clear, as I do see the confusion.